### PR TITLE
Feature/geopm

### PR DESCRIPTION
**Pull request by jlnav**

### CODE REVIEW

File: src/power_geopm/nrmpower_geopm.c [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+
+static nrm_client_t *client;
+static nrm_scope_t *scope;
+static nrm_sensor_t *sensor;
+
+static char *upstream_uri = "tcp://127.0.0.1";
+static int pub_port = 2345;
+static int rpc_port = 3456;
+
+#define assert__(x) for (; !(x); assert(x))
```
Comment (perarnau) [inline src/power_geopm/nrmpower_geopm.c]: why is this needed ?

File: src/power_geopm/nrmpower_geopm.c [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+{
+	int j, char_opt, err;
+	double freq = 1;
+	size_t i, MAX_SIGNAL_NAME_LENGTH = 55;
+	nrm_vector_t *signals = NULL;
+
+	// a vector of GEOPM signal names; will be pushed into by e.g.: -s
+	// DRAM_POWER -s CPU_POWER
+	assert(nrm_vector_create(&signals, MAX_SIGNAL_NAME_LENGTH) ==
+	       NRM_SUCCESS);
```
Comment (perarnau) [inline src/power_geopm/nrmpower_geopm.c:96]: you're pushing strings in this vector, the second argument should be sizeof(char *)

File: src/power_geopm/nrmpower_geopm.c [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+	char *signal_name, full_domain_name[NAME_MAX + 1];
+	for (i = 0; i < n_signals; i++) {
+		// pull out requested signal, convert to integer label
+		void *p;
+		nrm_vector_get(signals, i, &p);
+		signal_name = (char *)p;
+		// SignalNames[i] = signal_name;
+		strcpy(SignalNames[i], signal_name);
+		domain_type = geopm_pio_signal_domain_type(signal_name);
+		assert__(domain_type >= 0)
```
Comment (perarnau) [inline src/power_geopm/nrmpower_geopm.c]: use a test, and exit the program cleanly instead of a crash

File: src/power_geopm/nrmpower_geopm.c [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+
+		err = nrm_extra_create_name_ssu("nrm.geopm", suffix, 0,
+		                                &scope_name); // what index
+		                                              // should we use?
+		scope = nrm_scope_create(scope_name);
+
+		// signals like CPU_POWER belong to "package"
+		if (strcmp(suffix, "cpu") || strcmp(suffix, "package")) {
+			// lets use hwloc CPU indexes instead
+			numanode = hwloc_get_obj_by_type(
```
Comment (perarnau) [inline src/power_geopm/nrmpower_geopm.c]: no need for numa, there's a HWLOC_OBJ_SOCKET you can use

File: src/power_geopm/nrmpower_geopm.c (hunk 1 above)
Comment (jlnav) [reply src/power_geopm/nrmpower_geopm.c]: This was a "solution" to print an explanatory message on the assertion failing, but as you stated it's better to exit cleanly on the condition not being met

**Review by perarnau — APPROVED** [review]
LGTM
